src/Microsoft.DotNet.Wpf/src/PresentationBuildTasks/Microsoft/Build/Tasks/Windows/CreateTemporaryTargetAssemblyProject.cs

```diff
@@ -0,0 +1,389 @@
+        
+        //------------------------------------------------------
+        //
+        //  Public Properties
+        //
+        //------------------------------------------------------
+
+        #region Public Properties
+
+        /// <summary>
+        /// CurrentProject 
+        ///    The full path of current project file.
+        /// </summary>
+        [Required]
+        public string  CurrentProject
+        { get; set; }
+
+        /// <summary>
+        /// GeneratedCodeFiles
+        ///    A list of generated code files, it could be empty.
+        ///    The list will be added to the Compile item list in new generated project file.
+        /// </summary>
+        public ITaskItem[] GeneratedCodeFiles
+        { get; set; }
+
+        /// <summary>
+        /// CompileTypeName
+        ///   The appropriate item name which can be accepted by managed compiler task.
+        ///   It is "Compile" for now.
+        ///   
+        ///   Adding this property is to make the type name configurable, if it is changed, 
+        ///   No code is required to change in this task, but set a new type name in project file.
+        /// </summary>
+        [Required]
+        public string CompileTypeName
+        { get; set; }
+
+        /// <summary>
+        /// ReferencePath
+        ///    A list of resolved reference assemblies.
+        ///    The list will replace the original Reference item list in generated project file.
+        /// </summary>
+        public ITaskItem[] ReferencePath
+        { get; set; }
+
+        /// <summary>
+        /// ReferencePathTypeName
+        ///   The appropriate item name which is used to keep the Reference list in managed compiler task.
+        ///   It is "ReferencePath" for now.
+        ///   
+        ///   Adding this property is to make the type name configurable, if it is changed, 
+        ///   No code is required to change in this task, but set a new type name in project file.
+        /// </summary>
+        [Required]
+        public string ReferencePathTypeName
+        { get; set; }
+
+        [Output]
+        public string TemporaryTargetAssemblyProjectName 
+        { get; set; }
+
+        #endregion Public Properties
+  
+        //------------------------------------------------------
+        //
+        //  Private Methods
+        //
+        //------------------------------------------------------
+
+        #region Private Methods
+
+        //
+        // Remove specific items from project file. Every item should be under an ItemGroup.
+        //
+        private void RemoveItemsByName(XmlDocument xmlProjectDoc, string sItemName)
+        {
+            if (xmlProjectDoc == null || String.IsNullOrEmpty(sItemName))
+            {
+                // When the parameters are not valid, simply return it, instead of throwing exceptions.
+                return;
+            }
+
+            //
+            // The project file format is always like below:
```

Comment:
This comment is no longer accurate. For e.g., the SDK style projects no longer have `xmlns=""` in the `Project` element.